Library/Homebrew/exceptions.rb

```diff
@@ -524,9 +526,22 @@ def initialize(cause)
 
 # raised by safe_system in utils.rb
 class ErrorDuringExecution < RuntimeError
-  def initialize(cmd, args = [])
-    args = args.map { |a| a.to_s.gsub " ", "\\ " }.join(" ")
-    super "Failure while executing: #{cmd} #{args}"
+  def initialize(cmd, status:, stdout: nil, stderr: nil)
+    s = "Failure while executing; `#{cmd.shelljoin.gsub(/\\=/, "=")}` exited with #{status.exitstatus}."
+
+    if stdout
+      s << "==> Standard Output of failed command:\n"
+      s << stdout
+      s << "\n" unless stdout.end_with?("\n")
+    end
+
+    if stderr
+      s << "==> Standard Error of failed command:\n"
+      s << stderr
+      s << "\n" unless stderr.end_with?("\n")
+    end
```

Comment:
It probably would be good to expand `ErrorDuringExecution`'s test so that there isn't decrease in coverage here.